### PR TITLE
Disable _otelresource_ custom metric in non-AKS envs

### PR DESCRIPTION
**Pull request by trask**
test https://github.com/Azure/azure-sdk-for-java/pull/40312

### CODE REVIEW

File: smoke-tests/apps/OtelResourceCustomMetric/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/DefaultOtelResourceAttributesTest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+@UseAgent
+abstract class DefaultOtelResourceAttributesTest {
+
+  private static final List<String> EXPECTED_RESOURCE_ATTRIBUTES =
+      Arrays.asList(
+          "telemetry.sdk.language",
+          "service.name",
+          "service.instance.id",
```
Comment (trask) [inline smoke-tests/apps/OtelResourceCustomMetric/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/DefaultOtelResourceAttributesTest.java:34]: `service.instance.id` is coming from cloud role instance id: https://github.com/Azure/azure-sdk-for-java/blob/74213e109acd51d7a2a2972992dc4e13a466c159/sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/pipeline/TelemetryItemExporter.java#L165

Comment (heyams) [reply smoke-tests/apps/OtelResourceCustomMetric/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/DefaultOtelResourceAttributesTest.java:34]: 👍 it has nothing to do with ServiceInstanceIdResourceProvider. That makes sense.

**Comment by heyams** [comment]
can you please update [this comment ](https://raw.githubusercontent.com/microsoft/ApplicationInsights-Java/main/agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AiConfigCustomizer.java#:~:text=//%20enable%20%22io.opentelemetry.sdk.autoconfigure.internal.EnvironmentResourceProvider%22%20only.%20It%0A%20%20%20%20//%20enables%20all%20resource%20provider%20by%20default%0A%20%20%20%20properties.put(%0A%20%20%20%20%20%20%20%20%22otel.java.enabled.resource.providers%22%2C%0A%20%20%20%20%20%20%20%20%22io.opentelemetry.sdk.autoconfigure.internal.EnvironmentResourceProvider%22)%3B) mentioned that it excludes process attributes by default?

File: smoke-tests/apps/OtelResourceCustomMetric/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/DefaultOtelResourceAttributesTest.java (hunk 1 above)
Comment (heyams) [reply smoke-tests/apps/OtelResourceCustomMetric/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/DefaultOtelResourceAttributesTest.java:34]: should we start using ServiceInstanceIdResourceProvider then?

File: smoke-tests/apps/ConnectionStringOverrides/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/ConnectionStringOverridesTest.java [hunk 2]
```diff
@@ -29,9 +29,7 @@
 @UseAgent
 abstract class ConnectionStringOverridesTest {
 
-  @RegisterExtension
-  static final SmokeTestExtension testing =
-      SmokeTestExtension.builder().otelResourceAttributesEnvVar("key1=value1,key2=value2").build();
+  @RegisterExtension static final SmokeTestExtension testing = SmokeTestExtension.create();
```
Comment (trask) [inline smoke-tests/apps/ConnectionStringOverrides/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/ConnectionStringOverridesTest.java:32]: maybe we need an `AksConnectionStringOverridesTest` similar to the `AksRoleNameOverridesTest`?

Comment (trask) [reply smoke-tests/apps/ConnectionStringOverrides/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/ConnectionStringOverridesTest.java:32]: discussed, this is good as-is, I was mistaken

File: smoke-tests/apps/OtelResourceCustomMetric/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/DefaultOtelResourceAttributesTest.java (hunk 1 above)
Comment (trask) [reply smoke-tests/apps/OtelResourceCustomMetric/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/DefaultOtelResourceAttributesTest.java]: ServiceInstanceIdResourceProvider is currently in the otel sdk incubator module

**Review by heyams — APPROVED** [review]

**Pull request merged**